flask_cors/core.py

```diff
@@ -226,6 +226,13 @@ def set_cors_headers(resp, options):
         LOG.debug('CORS have been already evaluated, skipping')
         return resp
 
+    # Some libraries, like OAuthlib, set resp.headers to non Multidict
+    # objects (Werkzeug Headers work as well). This is a problem because
+    # headers allow repeated values.
+    if (not isinstance(resp.headers, Headers)
```

Comment:
Awesome catch that there are multiple types here. This makes me realize that checking the actual type is not necessarily what we want, rather we can duck-type this and make it a little more flexible for the future.

I'd suggest we change this to be:
```Python
if (not hasattr(resp.headers, 'add')):
```

Feel free to push back if that seems to nit-picky!